### ISSUE TITLE
Add comment on bucket calculations

### ISSUE THREAD

**Issue by ubamrein** [DP-3T/dp3t-sdk-backend#194]
Can you add a comment for describing what the `+1 - 1` is doing here? I think I get the `+1`, which will 'move' the `receivedAt` to the next day. But why the `-1`? Is it to put the time-tag to _just before midnight_?

Also, once again I'm not really sure this is correct with regard to the edge case: what happens if a set of keys get upserted (TIL: upsert) around midnight - depending on when the keys enter, they might be tagged on the next day or not.

Wouldn't it be better to make sure that all keys are from the past, then take the timestamp of the latest key to do this calculation, instead of the `nowMillis`?

_Originally posted by @ineiti in https://github.com/DP-3T/dp3t-sdk-backend/pull/188_

**Comment by ubamrein** [comment]
it is used to round received_at so that less information leaks if the DB leaks. It calculates the next bucket (+1) and subtracts 1 millisecond (-1) to be in the last millisecond of the actual received_at bucket.

First of all, we use a batch "upsert", which means one request inserts all keys in the same bucket (same received_at).

Second, the releaseBucketDuration is actually 2h. So in the worst case, the release gets delayed by two hours, which does not have a huge impact.

The thing is we want it to be nowMillis, since it is a "receivedAt". The Timestamp of the latest key can be anything in principle, also since we do not enforce any ordering on the receiving JSON (I'm not sure how the items are inserted into the list, I guess it is as it occurs in the JSON).

Long story short: I'll add a comment :)
